In [1]:
import pandas as pd
import numpy as np

print(pd.__version__)
print(np.__version__)


1.5.3
1.24.4


In [4]:
pip install seaborn

     ---------------------------------------- 0.0/294.9 kB ? eta -:--:--
     ------------------------- ------------ 194.6/294.9 kB 3.9 MB/s eta 0:00:01
     ------------------------------------ - 286.7/294.9 kB 2.9 MB/s eta 0:00:01
     -------------------------------------- 294.9/294.9 kB 2.6 MB/s eta 0:00:00
     ---------------------------------------- 0.0/8.1 MB ? eta -:--:--
     - -------------------------------------- 0.3/8.1 MB 8.9 MB/s eta 0:00:01
     -- ------------------------------------- 0.6/8.1 MB 7.5 MB/s eta 0:00:01
     ---- ----------------------------------- 0.9/8.1 MB 7.1 MB/s eta 0:00:02
     ----- ---------------------------------- 1.2/8.1 MB 7.4 MB/s eta 0:00:01
     -------- ------------------------------- 1.6/8.1 MB 8.1 MB/s eta 0:00:01
     ---------- ----------------------------- 2.2/8.1 MB 8.2 MB/s eta 0:00:01
     ------------- -------------------------- 2.7/8.1 MB 8.5 MB/s eta 0:00:01
     -------------- ------------------------- 3.0/8.1 MB 8.2 MB/s eta 

In [5]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt




In [6]:
df = pd.read_csv("sales.csv")
df.head()

Order ID Region  Product    Category  Sales (£)  Quantity      Supplier
0      1001  North   Laptop  Technology       1200         2      TechCorp
1      1004   West     Desk   Furniture        920         2   \tfficeMate
2      1002  South    Chair   Furniture        850         5  \tOfficeMate
3      1003   East  Monitor  Technology        600         3    \tOiceMate
4      1006  South    Table   Furniture        500         4      TechCorp

Removes spaces from column names

In [59]:
df.columns = df.columns.str.strip().str.lower().str.replace(' ','_')

In [66]:
df.head()

order_id region  product    category  sales_(£)  quantity      supplier
0      1001  North   Laptop  Technology       1200         2      TechCorp
1      1004   West     Desk   Furniture        920         2   \tfficeMate
2      1002  South    Chair   Furniture        850         5  \tOfficeMate
3      1003   East  Monitor  Technology        600         3    \tOiceMate
4      1006  South    Table   Furniture        500         4      TechCorp

In [60]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6 entries, 0 to 5
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   order_id   6 non-null      int64 
 1   region     6 non-null      object
 2   product    6 non-null      object
 3   category   6 non-null      object
 4   sales_(£)  6 non-null      int64 
 5   quantity   6 non-null      int64 
 6   supplier   6 non-null      object
dtypes: int64(3), object(4)
memory usage: 464.0+ bytes


In [61]:
df.isnull().sum()

order_id     0
region       0
product      0
category     0
sales_(£)    0
quantity     0
supplier     0
dtype: int64

In [62]:
df.describe()

order_id    sales_(£)   quantity
count     6.000000     6.000000   6.000000
mean   1003.500000   703.333333   4.333333
std       1.870829   366.969572   3.011091
min    1001.000000   150.000000   2.000000
25%    1002.250000   525.000000   2.250000
50%    1003.500000   725.000000   3.500000
75%    1004.750000   902.500000   4.750000
max    1006.000000  1200.000000  10.000000

In [ ]:
df['sales_(£)']

0    1200
1     920
2     850
3     600
4     500
5     150
Name: sales_(£), dtype: int64

Filter all orders from North region

In [67]:
df[df['region'] == 'North']

order_id region product    category  sales_(£)  quantity  supplier
0      1001  North  Laptop  Technology       1200         2  TechCorp
5      1005  North   Mouse  Technology        150        10   TechCrp

Region South

In [68]:
df[df['region'] == 'South']

order_id region product   category  sales_(£)  quantity      supplier
2      1002  South   Chair  Furniture        850         5  \tOfficeMate
4      1006  South   Table  Furniture        500         4      TechCorp

Group by

In [69]:
df.groupby('region')['sales_(£)'].sum()

region
East      600
North    1350
South    1350
West      920
Name: sales_(£), dtype: int64

Sorting Values by sales

In [71]:
df.sort_values(by='sales_(£)', ascending=False)

order_id region  product    category  sales_(£)  quantity      supplier
0      1001  North   Laptop  Technology       1200         2      TechCorp
1      1004   West     Desk   Furniture        920         2   \tfficeMate
2      1002  South    Chair   Furniture        850         5  \tOfficeMate
3      1003   East  Monitor  Technology        600         3    \tOiceMate
4      1006  South    Table   Furniture        500         4      TechCorp
5      1005  North    Mouse  Technology        150        10       TechCrp

The most sold product was the Mouse with sales of 150

In [74]:
df.sort_values(by= 'quantity', ascending=False)

order_id region  product    category  sales_(£)  quantity      supplier
5      1005  North    Mouse  Technology        150        10       TechCrp
2      1002  South    Chair   Furniture        850         5  \tOfficeMate
4      1006  South    Table   Furniture        500         4      TechCorp
3      1003   East  Monitor  Technology        600         3    \tOiceMate
0      1001  North   Laptop  Technology       1200         2      TechCorp
1      1004   West     Desk   Furniture        920         2   \tfficeMate